### ISSUE TITLE
Refactor local tar to using archive/tar pkg

### ISSUE THREAD

**Issue by pkieltyka** [pressly/sup#27]
See: https://github.com/pressly/sup/blob/master/tar.go#L36

**Comment by eduardonunesp** [comment]
:+1:

**Comment by pkieltyka** [comment]
@eduardonunesp feel free to submit a PR for this if you have the time :)

**Comment by eduardonunesp** [comment]
Sure, I'll take a look, need to learn more about archive/tar

**Comment by VojtechVitek** [comment]
I don't know if "tar/archive" will help us lot --- the same problem is on the end machine too, not just on localhost.

**Comment by eduardonunesp** [comment]
archive/tar can help if the localhost doesn't have the tar command right ? the Sup is using the minimum viable tar options now

**Comment by pkieltyka** [comment]
thats true. Having archive/tar at least helps the local machine, ie. what if someone uses this on Windows?

also, do we really need tar at all to transfer the data..? doesn't the library have some kind of scp equivalent to send a stream of file data?

**Comment by eduardonunesp** [comment]
A friend of mine, suggested to use rsync instead of tar. But i don't have idea how to replace :(

**Comment by VojtechVitek** [comment]
I'd love to use rsync, if available. But not all machines have that. We can maybe fall-back to tar, if necessary.

**Comment by eduardonunesp** [comment]
Agree, the local archive/tar looks good, in the case of Windows.

**Comment by pkieltyka** [comment]
I wish we could use just go libs for the upload instead of having to go down to cli calls to get a data stream for a bunch of files.

rsync is cool though since it does a file sync without a full upload

**Comment by VojtechVitek** [comment]
Yes, I'll do that for localhost. I can't do that for remote command though.
